### ISSUE TITLE
Add support for testing phyloreferences with different levels of strictness based on their status

### ISSUE THREAD

**Issue by gaurav** [phyloref/jphyloref#8]
The latest update of the Curation Tool (phyloref/curation-tool#66) adds a system for tracking the state of phyloreferences (phyloref/curation-tool#25). JPhyloref should use this information to determine which phyloreferences should be tested. Based on the list in phyloref/curation-tool#25:

- `pso:draft`: incomplete drafts -- should not be tested
- `pso:final-draft`: final drafts -- try to resolve, but mark as TODO if they fail
- `pso:under-review`: attempts to test this phyloreference have failed -- try to resolve, but mark as TODO if they fail
- `pso:submitted`: attempts to test this phyloreference has succeeded -- should be tested
- `pso:published`: have been/are to be published to the Clade Ontology -- should be tested
- `pso:retracted-from-publication`: were previously published but are now deprecated -- try to resolve, but mark as TODO if they fail

**Comment by hlapp** [comment]
> pso:retracted-from-publication: were previously published but are now deprecated -- try to resolve, but mark as TODO if they fail

Isn't this more or less the same as marking a term obsolete? Or are we not necessarily talking about classes in the Clade Ontology here, but PHYX files?

**Comment by gaurav** [comment]
> Isn't this more or less the same as marking a term obsolete? Or are we not necessarily talking about classes in the Clade Ontology here, but PHYX files?

I was thinking of the case our JSON-to-OWL conversion code or (more likely) our specifier matching code changes in such a way that an existing phyloreference no longer works as expected. It is intended to be a temporary state, after which the definition is corrected by a curator and the publication is returned to `pso:published`. Now that I think about it, the publication should probably be returned to `pso:under-review` in that case, so you're probably right that `pso:retracted-from-publication` is unnecessary.

I've filed an issue to detect and use `owl:deprecated` (#15) and to incorporate that into the Curation Tool (phyloref/curation-tool#81). Now that a mechanism for marking statuses of phyloreferences has been merged in #9, I'm going to close this issue -- we can track improvements to that system separately in different issues if needed.
